app/src/main/java/campus/tech/kakao/map/LocationAdapter.kt

```diff
@@ -0,0 +1,44 @@
+package campus.tech.kakao.map
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+
+class LocationAdapter(private val onLocationSelected: (Location) -> Unit) : ListAdapter<Location, LocationAdapter.LocationHolder>(
+    object : DiffUtil.ItemCallback<Location>() {
+        override fun areItemsTheSame(oldItem: Location, newItem: Location): Boolean {
+            return oldItem.title == newItem.title
+        }
+        override fun areContentsTheSame(oldItem: Location, newItem: Location): Boolean {
+            return oldItem == newItem
+        }
+    }) {
+    inner class LocationHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val titleTextView: TextView = itemView.findViewById(R.id.titleTextView)
+        val addressTextView: TextView = itemView.findViewById(R.id.addressTextView)
+        val categoryTextView: TextView = itemView.findViewById(R.id.categoryTextView)
+
+        init {
+            itemView.setOnClickListener {
+                val location = getItem(bindingAdapterPosition)
+                onLocationSelected(location)
+            }
+        }
+    }
```

Comment:
LocationViewHodler 가 LocationAdapter 의 것을 참조하지 않는 것이 좋습니다. (getItem 과 onLocationSelected) 
이를 inner class 가 아니고 parameter 로 받든가, 해서 처리하시는 것을 추천드려요.
```suggestion
    class LocationHolder(
        itemView: View, 
        private val onLocationSelected: (Int) -> Unit
    ) : RecyclerView.ViewHolder(itemView) {
        val titleTextView: TextView = itemView.findViewById(R.id.titleTextView)
        val addressTextView: TextView = itemView.findViewById(R.id.addressTextView)
        val categoryTextView: TextView = itemView.findViewById(R.id.categoryTextView)

        init {
            itemView.setOnClickListener {
                onLocationSelected(bindingAdapterPosition)
            }
        }
    }
```